include/seqan3/alignment/pairwise/edit_distance_unbanded.hpp

```diff
@@ -430,20 +430,23 @@ struct pairwise_alignment_edit_distance_unbanded
         {
             get<align_result_key::score>(res) = score();
         }
+
+        alignment_coordinate end;
         if constexpr (std::tuple_size_v<result_type> >= 3)
         {
-            get<align_result_key::end>(res) = end_coordinate();
+            end = end_coordinate();
+            get<align_result_key::end>(res) = end;
+        }
+
+        [[maybe_unused]] alignment_trace_matrix matrix = trace_matrix();
+        if constexpr (std::tuple_size_v<result_type> >= 4)
+        {
+            get<align_result_key::begin>(res) = alignment_begin_coordinate(matrix, end);
         }
-        // if constexpr (std::tuple_size_v<result_type> >= 4)
-        // { TODO
-        //     throw std::invalid_argument{"The current output setting is not supported!"};
-        // }
-        if constexpr (std::tuple_size_v<result_type> == 5)
+
+        if constexpr (std::tuple_size_v<result_type> >= 5)
         {
-            auto && [seq1_trace, seq2_trace] = alignment_trace(database, query, trace_matrix(), end_coordinate());
-            auto & [in1, in2] = get<align_result_key::trace>(res);
-            ranges::copy(seq1_trace, ranges::back_inserter(in1));
-            ranges::copy(seq2_trace, ranges::back_inserter(in2));
+            get<align_result_key::trace>(res) = alignment_trace(database, query, matrix, end);
```

Comment:
you don't need to cache `end`, as it is already part of the res. You can just use res.end_coordinate() here.